src/linear_algebra/bilinear_form.lean

```diff
@@ -779,6 +893,75 @@ end to_matrix
 
 end matrix
 
+namespace finsupp
+
+variables {α β : Type*}
+variables {M' : Type*} [add_comm_monoid M']
+variables {N : Type*} [has_zero N]
+
+lemma bilin_add_left' (A : α → β → N → (M' →+ R)) (p q : α →₀ M') (r : β →₀ N) :
+  (p + q).sum (λ i x, r.sum (λ j y, A i j y x))
+  = p.sum (λ i x, r.sum (λ j y, A i j y x)) + q.sum (λ i x, r.sum (λ j y, A i j y x)) :=
+begin
+  rw finsupp.sum_add_index,
+  { simp },
+  { simp }
+end
+
+lemma bilin_add_right' (A : α → β → N → (M' →+ R)) (p : α →₀ N) (q r : β →₀ M') :
+  p.sum (λ i x, (q + r).sum (λ j y, A i j x y))
+  = p.sum (λ i x, q.sum (λ j y, A i j x y)) + p.sum (λ i x, r.sum (λ j y, A i j x y)) :=
+begin
+  rw ← finsupp.sum_add,
+  congr,
+  ext i x,
+  rw finsupp.sum_add_index,
+  { simp },
+  { simp [mul_add] }
+end
+
+lemma bilin_smul_left' (A : α → β → N → (M →ₗ[R] R)) (a : R)
+  (p : α →₀ M) (q : β →₀ N) :
+  (a • p).sum (λ i x, q.sum (λ j y, A i j y x))
+  = a • p.sum (λ i x, q.sum (λ j y, A i j y x)) :=
+begin
+  let h : α → M →ₗ[R] R := λ i, q.sum (λ j y, A i j y),
+  have := @finsupp.sum_smul_index_linear_map' _ _ _ _ _ _ _ _ _ _ _ h,
+  convert this;
+  { ext i x,
+    simp [h] },
+end
+
+lemma bilin_smul_right' (A : α → β → N → (M →ₗ[R] R)) (a : R)
+  (p : α →₀ N) (q : β →₀ M) :
+  p.sum (λ i x, (a • q).sum (λ j y, A i j x y))
+  = a • p.sum (λ i x, q.sum (λ j y, A i j x y)) :=
+begin
+  rw finsupp.smul_sum,
+  congr,
+  ext i x,
+  exact finsupp.sum_smul_index_linear_map'
+end
+
+open bilin_form
+
+/-- A collection of `bilin_form R M`, parametrized by two indices in `α`, induces a bilinear form
+on `α →₀ M`.  This can be thought of as an infinite-dimensional version of `matrix.to_bilin_form`.
+-/
+noncomputable def to_bilin_form (A : α → α → (bilin_form R M)) :
```

Comment:
Perhaps a better name would be
```suggestion
noncomputable def bilin_form_of (A : α → α → (bilin_form R M)) :
```
since `finsupp.to_bilin_form` implies there is some finsupp being turned into a bilin_form, but there is not.

Reply:
Good idea!